module/zfs/mmp.c

```diff
@@ -205,49 +205,25 @@ typedef enum mmp_vdev_state_flag {
 	MMP_FAIL_WRITE_PENDING	= (1 << 1),
 } mmp_vdev_state_flag_t;
 
-static vdev_t *
-mmp_random_leaf_impl(vdev_t *vd, int *fail_mask)
+static void *
+mmp_list_next_wrap(list_t *list, void *object)
```

Comment:
No need for this to be `void *` we should be clear about the type.  I don't think it would be out of line to use an inline conditional for this either `?:`.

Reply:
Types fixed.

Re: inline conditional:  I might be misunderstanding you, but we might enter just the first conditional block, or both, or just the second one.  The "both" case is when object passed in, points to the last element of the list.